src/rich-log/domain.hh

```diff
@@ -88,24 +105,30 @@ struct domain_info
 ///
 ///     (this should be externally synchronized, otherwise it might create a race condition)
 ///
-#define RICH_LOG_DECLARE_DOMAIN(Name) RICH_LOG_DECLARE_DOMAIN_MV(Name, Debug)
-#define RICH_LOG_DECLARE_DOMAIN_MV(Name, min_verbosity)          \
-    namespace rlog::domains                                      \
-    {                                                            \
-    namespace Name                                               \
-    {                                                            \
-    enum : int                                                   \
-    {                                                            \
-        CompileTimeMinVerbosity = rlog::verbosity::min_verbosity \
-    };                                                           \
-                                                                 \
-    extern rlog::domain_info domain;                             \
-    }                                                            \
-    }                                                            \
+#define RICH_LOG_DECLARE_DOMAIN(Name) RICH_LOG_DECLARE_DOMAIN_EX(Name, Debug, extern)
+
+#define RICH_LOG_DECLARE_DOMAIN_MV(Name, MinVerbosity) RICH_LOG_DECLARE_DOMAIN_EX(Name, MinVerbosity, extern)
+
+#define RICH_LOG_DECLARE_DOMAIN_EX(Name, MinVerbosity, APIPrefix) \
```

Comment:
let's name it IMPL for now. the api is so unstable that it will break with a cough and I'd rather not end up with EX2, EX3, ... Or is this really something that is used frequently in user code?

Reply:
ah nvm, the API for this one might actually be kinda stable. DEFINE_EX will break on minor features

Reply:
I think it will be stable, yes. The API thing isn't great and usually shouldn't be necessary (because usually a DLL wouldn't want to export it's own domain for anyone else to use - just to access via the linked list) - but it _is_ necessary for the default domain when linking rlog dynamically